### DISCUSSION TITLE
Restart teleport agent and processes started by teleport were killed too

### DISCUSSION THREAD

**Discussion by thanhtunng** [gravitational/teleport#17112] [Q&A] [answered]
Sometimes i need to restart teleport agent, for example to renew ceritificate, and all processes i started using teleport were killed too (like run tomcat, java -jar &, ... ).
Is this inevitable or is there anyway to avoid killing processes when restarting a teleport agent?
Thanks a lot in advance!

**Comment by webvictim** [comment, marked as answer]
See this issue for some more discussion: https://github.com/gravitational/teleport/issues/2355

**Reply by thanhtunng** [reply]
Thanks a lot!! I tried out the solution in #2355 and it worked.

Solution 1: Add this config to teleport.service:
KillMode=process
Solution 2: Use reload instead of restarting
systemctl reload teleport
